script/test-robots

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+"""
+A script to run a test competition match on every robot in an submissions directory.
+
+A "broken" robot is any robot which:
+- Fails to reach the wait_start function within the 5 second timeout
+  set in wait_until_robots_ready, caused by:
+  a. The robot code throwing an uncaught exception before instantiating the Robot
+  b. The robot code taking over 5 seconds before instantiating the Robot
+- Fails to complete a 1 second match within the subprocess timout
+- Causes the simulator to exit with a non-zero exit code
+"""
+import re
+import sys
+import time
+import shutil
+import argparse
+import tempfile
+import subprocess
+from typing import List, Optional
+from pathlib import Path
+from zipfile import ZipFile
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+from controller_utils import NUM_ZONES  # isort:skip
+
+EXCESS_SUBPROCESS_TIMEOUT = 10
+
+
+def run_robot_match(
+    comp_match_script: Path,
+    archives_dir: Path,
+    tla: str,
+    failed_tlas: Optional[List[str]],
+    subprocess_timeout: Optional[int],
+) -> int:
+    try:
+        print(f"Running match for {tla}")  # noqa:T001
+
+        match_args = [
+            sys.executable,
+            str(comp_match_script),
+            str(archives_dir),
+            '0',  # no logs will be retained so a fixed match number is used
+        ]
+
+        match_args.append(tla)
+
+        match_args.extend(['-'] * (NUM_ZONES - 1))  # all other zones are empty
+
+        match_args.extend([
+            '--duration', '1',
+            '--no-record',
+        ])
```

Comment:
The order of the optional (`--named`) arguments doesn't matter -- they could easily be put first to avoid this break I think?
Alternatively (or additionally) we could spray the zones in or just append them:
```suggestion
        num_empty_zones = NUM_ZONES - 1
        match_args = [
            sys.executable,
            str(comp_match_script),
            str(archives_dir),
            '--duration', '1',
            '--no-record',
            '--',
            '0',  # no logs will be retained so a fixed match number is used
            tla,
        ] + (['-'] * num_empty_zones)
```